christfellowshipchurch/apollos-app
# [Group Chat] Add Message Group ability

# Related Todos
- [**🏕 Add Message Group button that opens group chat**](https://3.basecamp.com/3926363/buckets/17566548/todos/3114266024)

# Changes
- Adds a `<GroupChatButton>` component to Group pages, that only shows when you have access to the `GROUP_CHAT` feature flag on API
- Updates the existing `Channel` route and component to allow opening a group chat _or_ direct message chat (1df0ea4c93a83c80f8a879b4ec54f8caf90b345d)
- General cleanup as I explored the code, fixing lint warnings/errors and a typo component name `AvatarPlacholder`

# Tradeoffs/Caveats
No energy was spent refining design for the channel in the context of groups. This PR laser focuses on enabling the raw functionality and isn't the final form of how we'll want to initialize channels and/or query a users' role for it.

Even the button placement, which matches the shaped wireframes, is subject to change.

The Back button when in a group chat incorrectly takes you back to the Groups root page, instead of the individual group... logged that issue here: [**🏕 Make back button in group chat take you back to group page**](https://3.basecamp.com/3926363/buckets/17566548/todos/3123334086)

# Test Instructions
1. Make sure your local api is running `group-chat` branch
1. Log in as a user with access to the `GROUP_CHAT` feature flag
1. Open a group page
1. 🔸 Verify there is a new **Message Group** button near the breakout/zoom call ones
1. Tap the button
1. 🔸 Verify the chat opens for that group and that the groups' name is in the title

# Screenshots
|Gif|Screenshot|
|---|---|
|![2020-10-16 13 10 27](https://user-images.githubusercontent.com/1812955/96310073-9871a480-0fd4-11eb-80fe-1163854aa10e.gif)|<img src="https://user-images.githubusercontent.com/1812955/96310081-9dceef00-0fd4-11eb-87df-17e4f18fe3e3.jpg" width="500" />|
